`initialize_database` runs every time (at least with mysql 8)

### :speaking_head: Foreword

Thank for taking the time to fill this bug report fully. Without it we may not be able to fix the bug, and the issue may be closed without resolution.

## :ghost: Brief Description

`initialize_database` runs every time (at least with mysql 8)

## :pancakes: Cookbook version

10.0.2

## :woman_cook:  Chef-Infra Version

16.10.8

## :tophat: Platform details

CentOS 7.9

## Steps To Reproduce

Setup a service, using mysql 8

```
mysql_service 'foo' do
  port '3306'
  version '8.0'
  bind_address '127.0.0.1'
  data_dir node['foo']['mysql']['bar']
  initial_root_password node['foo']['mysql']['root_password']
  error_log node['foo']['mysql']['error_log']
  action [:create, :start]
end
```

On every run of chef-client, it will try and run the initialize database step:
```
    * bash[foo initial records] action run
      - execute "bash"
```

and you'll see a message like this in the mysql error log
```
2021-02-26T16:52:05.499384Z 0 [System] [MY-013169] [Server] /usr/sbin/mysqld (mysqld 8.0.23) initializing of server in progress as process 18946
2021-02-26T16:52:05.502231Z 0 [ERROR] [MY-010457] [Server] --initialize specified but the data directory has files in it. Aborting.
2021-02-26T16:52:05.502281Z 0 [ERROR] [MY-013236] [Server] The designated data directory /opt/mysql/ is unusable. You can remove all files that the server added to it.
2021-02-26T16:52:05.502416Z 0 [ERROR] [MY-010119] [Server] Aborting
2021-02-26T16:52:05.502829Z 0 [System] [MY-010910] [Server] /usr/sbin/mysqld: Shutdown complete (mysqld 8.0.23)  MySQL Community Server - GPL.
```


## :police_car: Expected behavior

This step should be skipped.  It's only supposed to run once.

## :heavy_plus_sign: Additional context

The step has a `not_if` options added to it:
https://github.com/sous-chefs/mysql/blob/master/libraries/mysql_service_base.rb#L89-L98
`not_if "/usr/bin/test -f #{new_resource.data_dir}/mysql/user.frm"`

For reasons I haven't yet discovered, this file doesn't exist on my system.  Creating it causes the expected behavior, and the `initialize_database` resource is skipped after the first run.  I suspect this may be a difference in mysql 5.7 and 8.0

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,6 +4,8 @@ All notable changes to this project will be documented in this file.
 
 ## Unreleased
 
+- Fix db initialization status correctly on MySQL 8.0
+
 ## 10.1.0 - *2021-03-11*
 
 - Add dependency on apparmor cookbook to fix #660

--- a/libraries/helpers.rb
+++ b/libraries/helpers.rb
@@ -250,6 +250,14 @@ EOSQL
       mysql_install_db_cmd
     end
 
+    def db_initialized?
+      if v80plus
+        ::File.exist? "#{data_dir}/mysql.ibd"
+      else
+        ::File.exist? "#{data_dir}/mysql/user.frm"
+      end
+    end
+
     def mysql_install_db_bin
       return "#{base_dir}/scripts/mysql_install_db" if platform_family?('omnios')
       return "#{prefix_dir}/bin/mysql_install_db" if platform_family?('smartos')

--- a/libraries/mysql_service_base.rb
+++ b/libraries/mysql_service_base.rb
@@ -92,7 +92,7 @@ module MysqlCookbook
           code init_records_script
           umask '022'
           returns [0, 1, 2] # facepalm
-          not_if "/usr/bin/test -f #{new_resource.data_dir}/mysql/user.frm"
+          not_if { db_initialized? }
           action :run
         end
       end

